### ISSUE TITLE
Custom Array.prototype interfering with for..in loops

### ISSUE THREAD

**Issue by GustavAT** [tangrams/tangram#374]
Hello,
I am not able to render any vector tiles on the map (altough they are valid, i checked them on another system), I found out that the custom Array-property that i am using is interfering with the for..in loop that is iterating over vector tiles.

Assume the following function:

```javascript
// concat two arrays
Array.prototype.extend = function (source) {
    source.forEach(function (el) { this.push(el); }, this);
};
```

As the for..in loop in javascript checks the entire prototype chain of an object being iterated, these can cause errors when iterating over arrays with custom properties, see example below:

```javascript
// inside renderStyle-function
for (var t in this.renderable_tiles) {
                var tile = this.renderable_tiles[t];
                
// tile is undefined when the loop reaches the "extend" property => crash
                if (tile.meshes[style] == null) {
                    continue;
                }
```

Below are some workarounds that i used:

- checking with *hasOwnProperty*

```javascript
for (var t in this.renderable_tiles) {
                if (!this.renderable_tiles.hasOwnProperty(t)) continue;
                var tile = this.renderable_tiles[t];
                ...
```
- using a normal for loop
```javascript
for (var i = 0; i < this.renderable_tiles; i++) {
                var tile = this.renderable_tiles[i];
                ...
```
- Object.keys
```javascript
Object.keys(this.renderable_tiles).forEach(funtion(key) {
                var tile = this.renderable_tiles[key];
                ...
```
- forEach
```
this.renderable_tiles.forEach(function(rec) {
              ...
```

Google Chrome 51.0.2704.103 m
tangram v0.8.2
Other frameworks used: Ext.Js 6.0.2

**Comment by bcamper** [comment]
Thanks - uses of `for..in` loops for arrays here are unintentional, probably leftover from those being objects in a previous implementation. We will take a look at updating.

**Comment by bcamper** [comment]
This is fixed in https://github.com/tangrams/tangram/commit/394bc5be7ff984ea1c70173a56b1ec480679d32a, and will be included in the next patch release, v0.9.1. I will close this issue when that release is out.

**Comment by GustavAT** [comment]
Thank you

**Comment by bcamper** [comment]
This has been released to production in v0.9.1
